From 4e63db1208cc34d8b1d6038324029551d7ee7ba4 Mon Sep 17 00:00:00 2001
From: Xufeng Zhang <xufeng.zhang@windriver.com>
Date: Thu, 18 Aug 2011 19:00:46 +0800
Subject: [PATCH 09/12] P1022ds: add the legerity node in device tree

Extracted from the QorIQ-NONDPAA-SDK-V1-20110429_ltib.iso vendor drop
linux-2.6.35-qoriq-fixed-the-spi-dts-bug.patch and
0111-P1022DS-DTS-Add-tdm-amp-support.patch.

Signed-off-by: Jerry Huang <Chang-Ming.Huang@freescale.com>

[Add the missed spi node for p1022ds_camp_core1_36bit.dts.]

Signed-off-by: Xufeng Zhang <xufeng.zhang@windriver.com>
---
 arch/powerpc/boot/dts/p1022ds.dts                  |   22 +++++++
 arch/powerpc/boot/dts/p1022ds_36bit.dts            |   22 +++++++
 arch/powerpc/boot/dts/p1022ds_camp_core1.dts       |   22 +++++++
 arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts |   62 ++++++++++++++++++++
 4 files changed, 128 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/boot/dts/p1022ds.dts b/arch/powerpc/boot/dts/p1022ds.dts
index ba2b273..2273cca 100644
--- a/arch/powerpc/boot/dts/p1022ds.dts
+++ b/arch/powerpc/boot/dts/p1022ds.dts
@@ -287,6 +287,28 @@
 					reg = <0x00700000 0x00900000>;
 				};
 			};
+		
+			legerity@0{
+				compatible = "fsl,espi-slic";
+				reg = <1>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <0>;
+				ch1-tx-slot = <0>;
+				ch2-rx-slot = <2>;
+				ch2-tx-slot = <2>;
+			};
+	
+			legerity@1{
+				compatible = "fsl,espi-slic";
+				reg = <2>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <4>;
+				ch1-tx-slot = <4>;
+				ch2-rx-slot = <6>;
+				ch2-tx-slot = <6>;
+			};
 		};
 
 		ssi@15000 {
diff --git a/arch/powerpc/boot/dts/p1022ds_36bit.dts b/arch/powerpc/boot/dts/p1022ds_36bit.dts
index 82af3a8..0ccbade 100644
--- a/arch/powerpc/boot/dts/p1022ds_36bit.dts
+++ b/arch/powerpc/boot/dts/p1022ds_36bit.dts
@@ -294,6 +294,28 @@
 					reg = <0x00700000 0x00900000>;
 				};
 			};
+		
+			legerity@0{
+				compatible = "fsl,espi-slic";
+				reg = <1>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <0>;
+				ch1-tx-slot = <0>;
+				ch2-rx-slot = <2>;
+				ch2-tx-slot = <2>;
+			};
+
+			legerity@1{
+				compatible = "fsl,espi-slic";
+				reg = <2>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <4>;
+				ch1-tx-slot = <4>;
+				ch2-rx-slot = <6>;
+				ch2-tx-slot = <6>;
+			};
 		};
 
 		ssi@15000 {
diff --git a/arch/powerpc/boot/dts/p1022ds_camp_core1.dts b/arch/powerpc/boot/dts/p1022ds_camp_core1.dts
index 2b81ca7..07612c1 100644
--- a/arch/powerpc/boot/dts/p1022ds_camp_core1.dts
+++ b/arch/powerpc/boot/dts/p1022ds_camp_core1.dts
@@ -206,6 +206,28 @@
 					reg = <0x00700000 0x00900000>;
 				};
 			};
+		
+			legerity@0{
+				compatible = "fsl,espi-slic";
+				reg = <1>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <0>;
+				ch1-tx-slot = <0>;
+				ch2-rx-slot = <2>;
+				ch2-tx-slot = <2>;
+			};
+
+			legerity@1{
+				compatible = "fsl,espi-slic";
+				reg = <2>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <4>;
+				ch1-tx-slot = <4>;
+				ch2-rx-slot = <6>;
+				ch2-tx-slot = <6>;
+			};
 		};
 
 		dma@c300 {
diff --git a/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts b/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts
index 9738654..3b938db 100644
--- a/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts
+++ b/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts
@@ -174,6 +174,68 @@
 			interrupts = <16 2>;
 		};
 
+		spi@7000 {
+			cell-index = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "fsl,espi";
+			reg = <0x7000 0x1000>;
+			interrupts = <59 0x2>;
+			interrupt-parent = <&mpic>;
+			espi,num-ss-bits = <4>;
+			mode = "cpu";
+
+			fsl_m25p80@0 {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "fsl,espi-flash";
+				reg = <0>;
+				linux,modalias = "fsl_m25p80";
+				spi-max-frequency = <40000000>; /* input clock */
+				partition@0 {
+					label = "u-boot-spi";
+					reg = <0x00000000 0x00100000>;
+					read-only;
+				};
+				partition@100000 {
+					label = "kernel-spi";
+					reg = <0x00100000 0x00500000>;
+					read-only;
+				};
+				partition@600000 {
+					label = "dtb-spi";
+					reg = <0x00600000 0x00100000>;
+					read-only;
+				};
+				partition@700000 {
+					label = "file system-spi";
+					reg = <0x00700000 0x00900000>;
+				};
+			};
+
+			legerity@0{
+				compatible = "fsl,espi-slic";
+				reg = <1>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <0>;
+				ch1-tx-slot = <0>;
+				ch2-rx-slot = <2>;
+				ch2-tx-slot = <2>;
+			};
+
+			legerity@1{
+				compatible = "fsl,espi-slic";
+				reg = <2>;
+				linux,modalias = "legerity";
+				spi-max-frequency = <8000000>;
+				ch1-rx-slot = <4>;
+				ch1-tx-slot = <4>;
+				ch2-rx-slot = <6>;
+				ch2-tx-slot = <6>;
+			};
+		};
+
 		dma@c300 {
 			#address-cells = <1>;
 			#size-cells = <1>;
-- 
1.7.0.4

